app/qml/ExternalResourceBundle.qml

```diff
@@ -39,17 +40,35 @@ Item {
             }
         }
 
-        property var imageSelected: function imageSelected(imagePath) {
-            var homePath  = featureForm.project ? featureForm.project.homePath : ""
-            var fileName = QgsQuick.Utils.getRelativePath(imagePath, homePath)
-            if (!fileName) {
-                fileName = __inputUtils.getFileName(imagePath)
-            }
-            if (!QgsQuick.Utils.fileExists(homePath, fileName)) {
-                __inputUtils.copyFile(imagePath, homePath + "/" + fileName)
-            }
-            externalResourceHandler.itemWidget.valueChanged(fileName, false)
+        property var confirmImage: function confirmImage(itemWidget, prefixToRelativePath, value) {
+          var newPath = __inputUtils.renameFile(prefixToRelativePath + "/" + value, customPrefix)
+          if (newPath) {
+            externalResourceHandler.itemWidget = itemWidget
+            var newCurrentValue = QgsQuick.Utils.getRelativePath(newPath, prefixToRelativePath)
+            externalResourceHandler.itemWidget.image.currentValue = newCurrentValue
+            externalResourceHandler.itemWidget.valueChanged(newCurrentValue, newCurrentValue === "" || newCurrentValue === null)
             externalResourceHandler.itemWidget = undefined
+          }
+        }
+
+        property var imageSelected: function imageSelected(imagePath) {
```

Comment:
How do we know externalResourceHandler.itemWidget is set already here? And why do we reset it to undefined at the end? I am a bit confused with the logic here...

It would be also good to add a bit of documentation what confirmImage() and imageSelected() are meant to do so one does not need to dig into QGIS code...